test/unit/resource-mapper-test.js

```diff
@@ -604,14 +606,26 @@ describe('ResourceMapper', () => {
 
     itMapsFile(mapper, 'a file with the port',
       {
-        path: `${rootPath}space/foo.html`,
+        path: `${rootPath}example.org/space/foo.html`,
         hostname: 'example.org'
       },
       {
         url: 'https://example.org:81/space/foo.html',
         contentType: 'text/html'
       })
   })
+
+  describe('A ResourceMapper instance for an HTTPS host with non-default port in a multi-host setup', () => {
+    const mapper = new ResourceMapper({ rootUrl: 'https://localhost:81/', rootPath, includeHost: true })
+
+    it('throws an error when there is an improper file path', (done) => {
+      mapper.mapFileToUrl({
+        path: `${rootPath}example.orgspace/foo.html`,
+        hostname: 'example.org'
+      }).then(() => done('Did not throw error'))
+      .catch(() => done())
```

Comment:
Nit: you can just write that as

```javascript
    it('throws an error when there is an improper file path', () => {
      return expect(mapper.mapFileToUrl({
        path: `${rootPath}example.orgspace/foo.html`,
        hostname: 'example.org'
      })).to.be.rejectedWith(new Error('error message'))
    })
```

and simultaneously test the error message. (`return` necessary because it's a promise)